Src/CSharpier/DocPrinter.cs

```diff
@@ -11,6 +11,9 @@ namespace CSharpier
     // taken from prettier 2.2.1 or so
     public static class DocPrinter
     {
+        [ThreadStatic]
+        private static Dictionary<string, PrintMode>? groupModeMap;
```

Comment:
It feels a little weird to have a ThreadStatic like this. I could also change DocPrinter to not be static to avoid this.